shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
+        return messageHeader.getDestinationFirstRep();
+    }
+
+    public static void setMSH5MessageDestinationComponent(
+            Bundle bundle, MessageHeader.MessageDestinationComponent receivingApplication) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        messageHeader.setDestination(List.of(receivingApplication));
+    }
+
+    // MSH-6 - Receiving Facility
+    public static Organization getMSH6Organization(Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return (Organization) messageHeader.getDestinationFirstRep().getReceiver().getResource();
+    }
+
+    public static void setMSH6Organization(Bundle bundle, Organization receivingFacility) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        Reference organizationReference = createOrganizationReference(bundle, receivingFacility);
+        MessageHeader.MessageDestinationComponent destination =
+                new MessageHeader.MessageDestinationComponent();
+        destination.setReceiver(organizationReference);
+        messageHeader.setDestination(List.of(destination));
+    }
+
+    // MSH-9 - Message Type
+    public static Coding getMSH9Coding(Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return messageHeader.getEventCoding();
+    }
+
+    public static void setMSH9Coding(Bundle bundle, Coding coding) {
+        var messageHeader = getMSHMessageHeader(bundle);
         messageHeader.setEvent(coding);
     }
 
-    public static MessageHeader findOrInitializeMessageHeader(Bundle bundle) {
-        var messageHeader = resourceInBundle(bundle, MessageHeader.class);
-        if (messageHeader == null) {
-            messageHeader = new MessageHeader();
-            bundle.addEntry(new Bundle.BundleEntryComponent().setResource(messageHeader));
+    // MSH-9.3 - Message Structure
+    public static String getMSH9_3Value(Bundle bundle) {
+        Coding coding = getMSH9Coding(bundle);
+        return coding.getDisplay();
+    }
+
+    public static void setMSH9_3Value(Bundle bundle, String value) {
+        Coding coding = getMSH9Coding(bundle);
+        coding.setDisplay(value);
+    }
+
+    // PID - Patient
+    public static Patient getPIDPatient(Bundle bundle) {
+        return resourceInBundle(bundle, Patient.class);
+    }
+
+    public static Patient createPIDPatient(Bundle bundle) {
+        Patient patient = new Patient();
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(patient));
+        return patient;
+    }
+
+    // PID-3 - Patient Identifier List
+    public static Identifier getPID3Identifier(Bundle bundle) {
+        Patient patient = getPIDPatient(bundle);
+        if (patient == null) {
+            return null;
+        }
+        return patient.getIdentifierFirstRep();
+    }
+
+    public static void setPID3Identifier(Bundle bundle, Identifier identifier) {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Moved to test package